### PR TITLE
[PSM Interop] Fix bootstrap generator interop test

### PR DESCRIPTION
**Pull request by zasweq**
This PR fixes the bootstrap generator interop test by making the node metadata flag dependent on version, which was causing a breakage previously as all bootstrap generator version's don't necessarily support the deexpiermentalized flag.

### CODE REVIEW

File: tools/run_tests/xds_k8s_test_driver/tests/bootstrap_generator_test.py [hunk 1]
```diff
@@ -205,6 +205,7 @@ def test_baseline_in_server_with_bootstrap_version(self, version, image):
             maintenance_port=self.server_maintenance_port,
             xds_host=self.server_xds_host,
             xds_port=self.server_xds_port,
+            bootstrap_version=version
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/tests/bootstrap_generator_test.py:208]: Style check failed with

> would reformat /var/local/git/grpc/tools/run_tests/xds_k8s_test_driver/tests/bootstrap_generator_test.py
> Oh no! 💥 💔 💥
> 1 file would be reformatted, 651 files would be left unchanged.

😄 

```suggestion
            bootstrap_version=version,
```

Comment (zasweq) [reply tools/run_tests/xds_k8s_test_driver/tests/bootstrap_generator_test.py]: Done. I guess when there's multiple arguments it wants full commas (I saw one argument doesn't have elsewhere).

**Comment by sergiitk** [comment]
Verifying existing tests:

- [ ] [grpc/java/v1.57.x/branch/xds_k8s_lb](https://source.cloud.google.com/results/invocations/8a81f0b2-9d8a-4aaf-b1c9-a425baa18dff)
- [ ] [grpc/java/v1.57.x/branch/psm-security](https://source.cloud.google.com/results/invocations/af508544-4b65-46cf-a1da-397a22ba7744)
- [ ] [grpc/java/v1.57.x/branch/xds_url_map](https://source.cloud.google.com/results/invocations/74479c5a-5d26-480e-ae75-27c9a43c229b)

**Comment by zasweq** [comment]
Discussed offline; let's hope these tests pass and merge this PR to unblock pluging into staging and prod cron jobs, and add true inequalities in templates + the TODO for version 10 to also trigger logic in templating file.

**Review by sergiitk — APPROVED** [review]

**Pull request merged**